drivers/ethernet/eth_stm32_hal.c

```diff
@@ -0,0 +1,417 @@
+/*
+ * Copyright (c) 2017 Erwin Rol <erwin@erwinrol.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define SYS_LOG_DOMAIN "dev/eth_stm32_hal"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ETHERNET_LEVEL
+#include <logging/sys_log.h>
+
+#include <kernel.h>
+#include <device.h>
+#include <misc/__assert.h>
+#include <misc/util.h>
+#include <errno.h>
+#include <stdbool.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <soc.h>
+#include <misc/printk.h>
+#include <clock_control.h>
+#include <clock_control/stm32_clock_control.h>
+
+#include "eth_stm32_hal_priv.h"
+
+static ETH_DMADescTypeDef dma_rx_desc_tab[ETH_RXBUFNB] __aligned(4);
+static ETH_DMADescTypeDef dma_tx_desc_tab[ETH_TXBUFNB] __aligned(4);
+static u8_t dma_rx_buffer[ETH_RXBUFNB][ETH_RX_BUF_SIZE] __aligned(4);
+static u8_t dma_tx_buffer[ETH_TXBUFNB][ETH_TX_BUF_SIZE] __aligned(4);
+
+
+static inline void disable_mcast_filter(ETH_HandleTypeDef *heth)
+{
+	__ASSERT_NO_MSG(heth != NULL);
+
+	u32_t tmp = heth->Instance->MACFFR;
+
+	/* disable multicast filtering */
+	tmp &= ~(ETH_MULTICASTFRAMESFILTER_PERFECTHASHTABLE |
+		 ETH_MULTICASTFRAMESFILTER_HASHTABLE |
+		 ETH_MULTICASTFRAMESFILTER_PERFECT);
+
+	/* enable receiving all multicast frames */
+	tmp |= ETH_MULTICASTFRAMESFILTER_NONE;
+
+	heth->Instance->MACFFR = tmp;
+
+	/* Wait until the write operation will be taken into account:
+	 * at least four TX_CLK/RX_CLK clock cycles
+	 */
+	tmp = heth->Instance->MACFFR;
+	HAL_Delay(ETH_REG_WRITE_DELAY);
```

Comment:
I'll lookup into it, that delay might not even be needed, I'll double check the reference manual how big it should be.